bindings/c/CMakeLists.txt

```diff
@@ -0,0 +1,56 @@
+cmake_minimum_required(VERSION 3.21)
+
+project(safetensors_c VERSION 0.3.0 LANGUAGES C CXX DESCRIPTION "Simple, safe way to store and distribute tensors")
+set(SAFETENSORS_PROJECT_NAME ${PROJECT_NAME})
+set(CMAKE_CXX_STANDARD 11)
+
+# Define options
+option(SAFETENSORS_BUILD_STATIC "Build the a static library" OFF)
+option(SAFETENSORS_BUILD_TESTS "Build the safetensors C-API tests" OFF)
+option(SAFETENSORS_BUILD_EXAMPLES "Build the safetensors C-API examples" OFF)
+
+# Define the c library header location
+set(SAFETENSORS_C_HEADERS include/safetensors/safetensors.h)
+add_library(safetensors_c_api INTERFACE)
+
+target_include_directories(safetensors_c_api INTERFACE
+    $<BUILD_INTERFACE:${CMAKE_CURRENT_LIST_DIR}/include>
+    $<BUILD_INTERFACE:${CMAKE_CURRENT_LIST_DIR}/include>
+)
+
+# Build up Rust part
+include(FetchContent)
+
+FetchContent_Declare(
+    Corrosion
+    GIT_REPOSITORY https://github.com/corrosion-rs/corrosion.git
+    GIT_TAG v0.4 # Optionally specify a commit hash, version tag or branch here
+)
+FetchContent_MakeAvailable(Corrosion)
```

Comment:
Seeing that Corrosion [installs find_package targets](https://github.com/corrosion-rs/corrosion/blob/123be1e3d8170c86e121392e8bffa4def7dc3447/CMakeLists.txt#L67-L73), it would be nice to add explicit `FIND_PACKAGE_ARGS` (helpful for distributions)